rust/src/text_encoding.rs

```diff
@@ -0,0 +1,42 @@
+use automerge as am;
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum TextEncoding {
+    UnicodeCodePoint,
+    Utf8CodeUnit,
+    Utf16CodeUnit,
+    GraphemeCluster,
+}
+
+impl From<TextEncoding> for am::TextEncoding {
+    fn from(value: TextEncoding) -> Self {
+        match value {
+            TextEncoding::UnicodeCodePoint => am::TextEncoding::UnicodeCodePoint,
+            TextEncoding::Utf8CodeUnit => am::TextEncoding::Utf8CodeUnit,
+            TextEncoding::Utf16CodeUnit => am::TextEncoding::Utf16CodeUnit,
+            TextEncoding::GraphemeCluster => am::TextEncoding::GraphemeCluster,
+        }
+    }
+}
+
+impl From<am::TextEncoding> for TextEncoding {
+    fn from(value: am::TextEncoding) -> Self {
+        match value {
+            am::TextEncoding::UnicodeCodePoint => TextEncoding::UnicodeCodePoint,
+            am::TextEncoding::Utf8CodeUnit => TextEncoding::Utf8CodeUnit,
+            am::TextEncoding::Utf16CodeUnit => TextEncoding::Utf16CodeUnit,
+            am::TextEncoding::GraphemeCluster => TextEncoding::GraphemeCluster,
+        }
+    }
+}
+
+impl std::fmt::Display for TextEncoding {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::UnicodeCodePoint => write!(f, "UnicodeCodePoint"),
+            Self::Utf8CodeUnit => write!(f, "Utf8CodeUnit"),
+            Self::Utf16CodeUnit => write!(f, "Utf16CodeUnit"),
```

Comment:
```suggestion
            Self::Utf16CodeUnit => write!(f, "UTF-16CodeUnit"),
```

Reply:
addressed in https://github.com/automerge/automerge-swift/commit/e7d18e95a1eb6b07b55efd35046ed759004c8172